modules/promotion/src/Entity/Promotion.php

```diff
@@ -176,6 +176,16 @@ public function getOffer() {
   /**
    * {@inheritdoc}
    */
+  public function setOffer($target_plugin_id, array $target_plugin_configuration) {
```

Comment:
We should pass an Offer plugin, and then the method knows how to extract the plugin ID and configuration